proto/message.py

```diff
@@ -568,11 +573,21 @@ def __init__(
                 # See related issue
                 # https://github.com/googleapis/python-api-core/issues/227
                 if isinstance(value, dict):
-                    keys_to_update = [
-                        item
-                        for item in value
-                        if not hasattr(pb_type, item) and hasattr(pb_type, f"{item}_")
-                    ]
+                    if _message is None:
```

Comment:
For this code block (and other similar blocks in this PR) please add an explicit comment for each if branch which protobuf version it belongs to (3.20 or 4.21 case).

Reply:
Done